src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -103,34 +125,42 @@ def to_forecast(
 
 
 class TabularEstimator(Estimator):
-    def __init__(self, freq: str, prediction_length: int, **kwargs) -> None:
+    def __init__(self, freq: str, prediction_length: int, use_lag=False, lags=[], **kwargs) -> None:
         super().__init__()
         self.task = task
         self.freq = freq
         self.prediction_length = prediction_length
         default_kwargs = {
+            "eval_metric": "mean_absolute_error",
             "excluded_model_types": ["KNN", "XT", "RF"],
             "presets": [
-                "high_quality_fast_inference_only_refit",
-                "optimize_for_deployment",
+                "high_quality_fast_inference_only_refit", "optimize_for_deployment"
             ],
-            "eval_metric": "mean_absolute_error",
         }
         self.kwargs = {**default_kwargs, **kwargs}
+        self.use_lag = use_lag
+        if not self.use_lag:
+            self.lags = []
+        else:
+            self.lags = sorted(list(set(lags))) if lags else list(get_lags_for_frequency(self.freq))
 
     def train(self, training_data: Dataset) -> TabularPredictor:
+        # every time there is only one time series passed
+        # list(training_data)[0] is essentially getting the only time series
         dfs = [
-            get_prediction_dataframe(to_pandas(entry))
+            get_prediction_dataframe(series=to_pandas(entry),
+                                     prediction_length=self.prediction_length,
+                                     use_lag=self.use_lag,
+                                     context_length=len(self.lags), train=True)
```

Comment:
So I think the lags are supposed to indicate exactly which previous data points from the time series should be used as features. Here, however, you're just taking the amount of them, and (if I understand correctly), let's call it `n`, use the `n` previous data points as features. Shouldn't you pass `self.lags` here, and use those to construct the dataframe?

Reply:
Yeah, it is a mistake. Will correct this accordingly. Thanks!